plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/ext/provider/ApiEndpointEnvVariableProvider.java

```diff
@@ -30,6 +32,11 @@
 
     @Override
     public String get() {
+        String apiEndpointEnvVar = System.getenv(DockerInstanceRuntimeInfo.API_ENDPOINT_URL_VARIABLE);
```

Comment:
Hello @garagatyi, sorry for answering late to your questions but I became father for the second time yesterday and I'm taking a few days off (not completely off though 😄)

By the way to aswer your question:

* Fonctionality: Add the ability to set the property `machine.docker.che_api.endpoint` using an environment variable (`CHE_API_ENDPOINT`). 
* How it will be used: `export CHE_API_ENDPOINT=<...> && bin/che.sh run` or `docker run -e CHE_API_ENDPOINT="<...>" ... codenvy/che`

I think you suggest me to set that variable to resolve problems running Che on Docker for Mac beta. I tried to set that variable but Che didn't parse it. So I've fixed it. It's useful in situations where Che fails retrieving the server IP.